pkg/userdata/rhel/provider.go

```diff
@@ -186,10 +186,6 @@ write_files:
     hostnamectl set-hostname {{ .MachineSpec.Name }}
     {{ end }}
 
-    subscription-manager clean
```

Comment:
Using subscription clean was recommended from RedHat support.

Reply:
This line is the root cause of our issue. Without cleaning the subsequent `subscription-manager register` command would fail the second time the script is executed.

I think that we should have added this not to leak registrations as mentioned [here](https://access.redhat.com/solutions/228883):
```
# subscription-manager remove --all
# subscription-manager unregister
# subscription-manager clean
```

but this would unregister and register the system all the times the script is run.  I think the solution proposed here is cleaner.

Reply:
I see, actually this is what I was advised to do when the subscription manager didn't work.

Reply:
ok, anyway the [rh_subscription](https://github.com/canonical/cloud-init/blob/3c551f6ebc12f7729a2755c89b19b9000e27cc88/cloudinit/config/cc_rh_subscription.py) module is actively maintained also by RH folks, I think it should be ok.

Reply:
> But actually this is not happening. I have tested it and I saw that whether the node is registered or not the subscription is recreated.

Well, there is a test based on `subsctiption-manager identity` command:
https://github.com/canonical/cloud-init/blob/3c551f6ebc12f7729a2755c89b19b9000e27cc88/cloudinit/config/cc_rh_subscription.py#L56
https://github.com/canonical/cloud-init/blob/master/tests/unittests/test_rh_subscription.py#L43

Maybe what you meant is that it does not perform a remote check with Red Hat Subscription Management API, but this does not seem to be necessary for the purpose of this PR.